fileglancer_central/usercontext.py

```diff
@@ -0,0 +1,80 @@
+import os
+import pwd
+
+from loguru import logger
+
+
+class UserContext:
+    def __init__(self, username):
+        self.username = username
+        self._uid = os.getuid()
+        self._gid = os.getgid()
+        self._gids = os.getgrouplist(pwd.getpwuid(self._uid).pw_name, self._gid)
+        self._user = None
+
+    def __enter__(self):
+        logger.debug(f"Prepare user context for {self.username}")
+        user = pwd.getpwnam(self.username)
+
+        uid = user.pw_uid
+        gid = user.pw_gid
+        gids = os.getgrouplist(self.username, gid)
+        try:
+            os.setegid(gid)
+        except Exception as e:
+            logger.error(f"Failed to set the effective gid: {e}")
+            raise e
+
+        try:
```

Comment:
@krokicki on my mac I ran into an exception when I tried to restore the groups - the number of groups my root has on mac is larger than the number allowed to set. So I added this check just to see if this happens